packages/web-config-server/src/apiV1/legacyReport.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import ViewRouteHandler from './view';
+import { DashboardItemPermissionChecker } from './permissions';
+
+export default class extends ViewRouteHandler {
+  static PermissionsChecker = DashboardItemPermissionChecker;
+
+  fetchReport = async ({ itemCode }) => {
+    const dashboardItem = await this.models.dashboardItem.findOne({
+      code: itemCode,
+      legacy: true,
+    });
+    if (!dashboardItem) {
+      throw new Error(`No dashboard item exists with the code ${itemCode}`);
+    }
+    const legacyReport = await this.models.legacyReport.findOne({
+      code: dashboardItem.report_code,
+    });
+    if (!legacyReport) {
+      throw new Error(`No legacy report exists with id ${dashboardItem.report_code}`);
+    }
+    return {
+      viewJson: dashboardItem.config,
```

Comment:
@chris-bes when you get to integrating this with Tupaia, if we can remove the need for the front end to receive `viewJson` at this point (as it should already have it from the first `/dashboards` request), then we can change this lookup to use the `reportCode` rather than the `itemCode`, and remove all need for the `/legacyReport` endpoint to query the `dashboardItem` table (which would more appropriately restrict its responsibility).

@EMcQ-BES has kept the `viewJson` in the response here in case there's too much to unpick to remove it.

Reply:
That's a great idea @edmofro ! I agree it'd be better not to return `viewJson` here. 

However, I just have 1 concern that even if we don't need to return `viewJson`, we still need to send in the `itemCode` because if we only restrict it to `legacy_report` look up here, we won't be able to check for permission since all the permission columns are in the `dashboard_relation`. And it may not be possible to map the `report_code` back to a unique `dashboard_relation` because 1 `legacy_report` may be used in multiple `dashboard_item`? So I feel it's still necessary to pass the `itemCode` (and also `dashboardCode`) to this endpoint purely for permission check purpose. I can think of 2 options:

1. Have only 1 endpoint `dashboardItem/:itemCode/report?dashboardCode=xxx` which will check for the right permissions and also fetch either legacy report or new report based on `legacy`. So the front end just need to call this endpoint.
Or
2. At least add another `permission_group` column into `legacy_report` (similar to `report` table) so that we can check the permission (it will not be as strict as how we check in dashboard_relation)

I'm open to any other suggestions as well!

Reply:
Oh right I see the issue. I don't think we should go for option 2., as it seems like unnecessary extra work for a legacy table. So how about another option:

3. Consistent endpoint used by both, `/report/:reportCode?itemCode=yyy&legacy=true`

Points of difference:
- Term `reportCode` rather than `dashboardCode` (dashboards are now what we formerly called dashboard groups)
- `reportCode` moved to endpoint segment rather than url param (it's mandatory when fetching from that endpoint)
- `itemCode` moved to query param, as it's only required for legacy reports so could one day be dropped
- `legacy` status passed through from front end in query param, so that the back end route handler could process a non-legacy report without having to go to the database at all (doesn't have to re-look up the dashboard item, just passes request straight off to report server like a well behaved orchestration server)

What do you think?

Reply:
> Term reportCode rather than dashboardCode (dashboards are now what we formerly called dashboard groups)

Oh I actually did mean `dashboardCode` in this case, because a `dashboard_item` can be linked to multiple dashboards, so just `itemCode` itself won't be sufficient to fetch a unique `dashboard_relation`.

Agreed with all other points! The only reason why I chose passing only `itemCode` but not `reportCode` was we can get the `reportCode` from `dashboard_item`, and it might mean we're sending redundant information. Also to prevent the case that the passed in `itemCode` and `reportCode` don't actually map together (ie: `itemCode` is a total different one that don't use the passed in `report_code`, but maybe we can also validate this in PermissionChecker?). So I was kinda leaning that `dashboard_item` is the source of truth. But if you think we don't have to worry about that then happy with your suggestion 👍

Reply:
Aaah right, sorry I didn't take enough time understanding the reason for `dashboardCode`, that makes sense.

But yes confirming I'm happy for report data fetches to skip the validation of the dashboard item, given that reports have a permission check on them themselves it's not a security issue (perhaps could catch a front end bug, but it'd be a pretty dumb bug ;-)). The legacy reports should have a permission check that does use the dashboard item code and dashboard code, but hopefully one day the legacy endpoint can be dropped entirely (maybe in 2035)